proposals/infra/INF-NNNN-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,126 @@
+<!-- {% raw %} -->
+
+# Support for in-process, Out-of-process, or both for the c style compiler api
+
+* Proposal: [NNNN](INF-NNNN-inproc-outofproc-compiler-api-support.md)
```

Comment:
It was mentioned in a previous PR that the name of this proposal doesn't fully match the contents.  Originally this was a discussion type proposal but it reads like a decision has been made to choose out of process.  There will be a design meeting that may change the contents here, so I think I can keep this name.  Any objections to this?

Reply:
All proposals in this repo are "infra" ones from hlsl specs, so there's no need for the infra prefix or for adding the infra directory.

Reply:
I'll fix this up.  I wasn't sure.

Reply:
fixed!